### PR TITLE
[FEAT] Consumer support for specifying num_replicas

### PR DESCRIPTION
**Pull request by aricart**

### CODE REVIEW

File: nats-base-client/types.ts [hunk 1]
```diff
@@ -2392,7 +2397,8 @@ export interface ConsumerUpdateConfig {
    */
   "max_bytes"?: number;
   /**
-   * When set do not inherit the replica count from the stream but specifically set it to this amount
+   * When set do not inherit the replica count from the stream but specifically set it to this amount.
+   * Min is 1, Max is 5
```
Comment (kozlovic) [inline nats-base-client/types.ts:2401]: I don't think library should enforce a max, and not sure even server does at this time (there may be configuration to enforce a max, but that should be left to the server/config and not at the library level)

**Review by kozlovic — APPROVED** [review]
LGTM, but would remove the note about replicas max value.

File: nats-base-client/types.ts (hunk 1 above)
Comment (aricart) [reply nats-base-client/types.ts]: done.

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**